test/app-luatest/fiber_test.lua

```diff
@@ -96,3 +96,21 @@ g.test_gh_10196_no_hang_on_self_join = function()
         message = 'cannot join itself',
     })
 end
+
+g.test_fiber_set_system = function()
```

Comment:
- I don't see much point in testing this internal function.
- Better squash this commit with the next one, which uses the new functionality.

Reply:
Done.